### ISSUE TITLE
Tertiary button `is-pressed` style insufficient contrast ratio on hover

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68535]
### Description

Discovered while working on https://github.com/WordPress/gutenberg/pull/68531 

The tertiary button style variant can have (like all other variants) a `pressed` state. With this styling, the button background becomes a dark grrey. However, when hovering the button, the button text becomes blue. It is barely visible:

![Image](https://github.com/user-attachments/assets/5b424647-1160-485f-aa27-1cbb2f0a172c)

This appears to happen only with the `tertiary` variant. For the other variants, the text stays whie on hover.

### Step-by-step reproduction instructions

- Go to https://wordpress.github.io/gutenberg/?path=/story/components-button--tertiary&args=aria-pressed:true
- Make sure the aria-pressed prop is true.
- Hover the button.
- Observe the text becomes blue on a dark gray background.
- Alternatively, go to the Post editor.
- Enable the preference `Show button text labels` from the top bar Options > Preferences > Accessibility.
- Make sure the Settings panel is open by clicking 'Settings' in the top bar.
- Hover the Settings button and observe the text becomes blue on a dark gray background.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by patil-vipul** [comment]
@afercia I've verified the issue for the `tertiary` variant for the `hover` state as shown below.

![Image](https://github.com/user-attachments/assets/4ac78815-861b-493b-a27b-2af85053fe2c)


However, the `pressed` or `clicked` state changes the button style to outline.

![Image](https://github.com/user-attachments/assets/615f2e20-7a02-4ad4-b765-2d8f54e820d1)

I've tested this on the latest version of WordPress (6.7.1). I'll raise the PR for the same.

**Comment by yogeshbhutkar** [comment]
I believe the following line is overriding the color property set by `is-pressed` rule:
https://github.com/WordPress/gutenberg/blob/ef7afefc5988eaab30da0025aa48bfbddb09a79f/packages/components/src/button/style.scss#L172

This shouldn't be applied when `is-pressed` is set as it already applies the color and background properties.
https://github.com/WordPress/gutenberg/blob/ef7afefc5988eaab30da0025aa48bfbddb09a79f/packages/components/src/button/style.scss#L351-L358

**Comment by yogeshbhutkar** [comment]
Thank you, @patil-vipul, for testing the issue and apologies for submitting this PR after you had already committed to raising it. While reviewing the bug, I noticed a simple one-line fix and decided to create this quick PR.

That said, I’m open to exploring better solutions and would love to collaborate further to address this issue effectively.

**Comment by t-hamano** [comment]
I identified that this issue occurred in #67325. In other words, this line:

https://github.com/WordPress/gutenberg/blob/bdb16796eb85051b9d5d7b00fcb773973642a5e7/packages/components/src/button/style.scss#L172

#67325 seems to have only attempted to improve the hover style of the secondary variation, so I'm not sure why there are changes to the tertiary variation. It would be good to look into why this color is needed in the first place.

This issue first appeared in 6.8, so let's try to fix it in the 6.8 release.
